feat(unzipWith): add unzipWith (#113)
toss/es-toolkit: docs/reference/array/unzipWith.md

# unzipWith

> A function that splits or transforms an input array into a new array. This function takes a nested array and returns a new array by either splitting the elements into separate arrays or using an optional iteratee function to transform the unzipped elements.

Reviewer: ```suggestion
Unzips an array of arrays, applying an `iteratee` function to regrouped elements.
```

## Signature

```typescript

> function unzipWith<T>(target: readonly T[][] | null | undefined): T[][];
> function unzipWith<T, R>(target: readonly T[][] | null | undefined, iteratee: (...args: T[]) => R): R[];

Reviewer: ```suggestion
function unzipWith<T, R>(target: T[][], iteratee: (...args: T[]) => R): R[];
```

```

### Parameters

> - `target`(`T[][] | null | undefined`): The nested array to split or transform. If the array is null or undefined, an empty array is returned.
> - `iteratee` (`(...args: T[]) => R`, 선택 사항): A function to transform the unzipped elements. If provided, the iteratee is called with the corresponding elements from the inner arrays, and its return value is included in the result.

Reviewer: ```suggestion
- `target`(`T[][]`): The nested array to unzip. This is an array of arrays, where each inner array contains elements to be unzipped.
- `iteratee` (`(...args: T[]) => R`): A function to transform the unzipped elements.
```
